pkgs/build-support/docker/default.nix

```diff
@@ -281,9 +281,13 @@ rec {
       '');
     in
     runCommand name {} ''
-      mkdir -p $out
       cd ${result}
-      cp layer.tar json VERSION $out
```

Comment:
Why was this added in the first place?
Your fix is probably fine for `exportImage`, but this completely breaks the use-case of just running a container (virtual-machine, but same difference) and copying something out of it. I'm not too deep into docker or nixpkgs's dockerTools, but the need to output a tar archive and these two files from this seems pretty arbitrary.

Reply:
It was introduced here https://github.com/NixOS/nixpkgs/pull/63865/files#r613386178

Reply:
It seems that these files are part of the buildLayered-style layer derivation output format.
`1.0` might be a good default `VERSION`
I don't know what should be in `json`, but perhaps that can be filled in as well.

We should try to unify `dockerTools`, not add more corner cases.

Reply:
If we're happy to change the behaviour from creating a tarball at $out to $out/layer.tar then it seems reasonable to just touch them and leave them empty? To maintain previous behaviour with a tarball at `$out` we could wrap this result in another derivation that copies out `layer.tar`. This would leave `runWithOverlay` unmodified, but double the disk usage of `exportImage`.

Reply:
I've removed this logic because it wasn't in the original definition, the name suggests that it's a fairly generic function and this uncluttering logic was too specific, yet unnecessary when making `createEmptyImage` write to a non-$out location.